test/parallel/test-repl-underscore.js

```diff
@@ -0,0 +1,157 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const repl = require('repl');
+const stream = require('stream');
+
+testSloppyMode();
+testStrictMode();
+testResetContext();
+testMagicMode();
+
+function testSloppyMode() {
+  const r = initRepl(repl.REPL_MODE_SLOPPY);
+
+  // cannot use `let` in sloppy mode
+  r.write(`_;          // initial value undefined
+          var x = 10;  // evaluates to undefined
+          _;           // still undefined
+          y = 10;      // evaluates to 10
+          _;           // 10 from last eval
+          _ = 20;      // explicitly set to 20
+          _;           // 20 from user input
+          _ = 30;      // make sure we can set it twice and no prompt
+          _;           // 30 from user input
+          y = 40;      // make sure eval doesn't change _
+          _;           // remains 30 from user input
+          `);
+
+  assertOutput(r.output, [
+    'undefined',
+    'undefined',
+    'undefined',
+    '10',
+    '10',
+    'expression assignment to _ now disabled',
+    '20',
+    '20',
+    '30',
+    '30',
+    '40',
+    '30'
+  ]);
+}
+
+function testStrictMode() {
+  const r = initRepl(repl.REPL_MODE_STRICT);
+
+  r.write(`_;          // initial value undefined
+          var x = 10;  // evaluates to undefined
+          _;           // still undefined
+          let _ = 20;  // use 'let' only in strict mode - evals to undefined
+          _;           // 20 from user input
+          _ = 30;      // make sure we can set it twice and no prompt
+          _;           // 30 from user input
+          var y = 40;  // make sure eval doesn't change _
+          _;           // remains 30 from user input
+          function f() { let _ = 50; } // undefined
+          f();         // undefined
+          _;           // remains 30 from user input
+          `);
+
+  assertOutput(r.output, [
+    'undefined',
+    'undefined',
+    'undefined',
+    'undefined',
+    '20',
+    '30',
+    '30',
+    'undefined',
+    '30',
+    'undefined',
+    'undefined',
+    '30'
+  ]);
```

Comment:
Care to explain why `expression assignment to _ now disabled` is not printed in strict/magic modes? I think I'm missing something.